src/main/scala/edu/berkeley/ce/rockslicing/LoadBalancer.scala

```diff
@@ -0,0 +1,323 @@
+                                      "seed joints")
+    }
+    seedJoints
+  }
+
+  /**
+    * Finds the best set of processor joints to divide initial volume into approximately equal volumes
+    * @param joints List to prepend processor joints to during each recursion
+    * @param origin Origin which the joint is in reference to
+    * @param center Center of the joint plane - start with lower left corner of bounding box
+    * @param initialVolume Intial volume that needs to be sub-divided
+    * @param desiredVolume Desired volume for each piece cut by processor joints
+    * @return List of processor joints that will be used to seed initial RDD
+    */
+  @tailrec
+  private def findProcessorJoints(joints: Seq[Joint],
+                                  origin: (Double, Double, Double),
+                                  center: (Double, Double, Double), initialVolume: Block,
+                                  desiredVolume: Double):
+                                  Seq[Joint] = {
+    // Calculate bounding box
+    val vertices = initialVolume.findVertices.values.flatten
+    val x_max = (vertices.map{ case vertex => vertex._1}).max
+    val x_min = (vertices.map{ case vertex => vertex._1}).min
+    val y_max = (vertices.map{ case vertex => vertex._2}).max
+    val y_min = (vertices.map{ case vertex => vertex._2}).min
+    val z_max = (vertices.map{ case vertex => vertex._3}).max
+    val z_min = (vertices.map{ case vertex => vertex._3}).min
+    val boundingBox = (x_min, y_min, z_min, x_max, y_max, z_max)
+
+    // Calculate diagonal length
+    val x_diff = boundingBox._4 - boundingBox._1
+    val y_diff = boundingBox._5 - boundingBox._2
+    val z_diff = boundingBox._6 - boundingBox._3
+    val diagonalLength = math.sqrt(x_diff*x_diff + y_diff*y_diff + z_diff*z_diff)
+    val normal = linalg.normalize(DenseVector[Double](x_diff, y_diff, z_diff))
+
+    val joint = Joint((normal(0), normal(1), normal(2)), origin, center, phi = 0.0, cohesion = 0.0,
+                      shape = Nil, processorJoint = true)
+    val blocks = initialVolume.cut(joint)
+    val nonRedundantBlocks = blocks.map { case block @ Block(center, _) =>
+      Block(center, block.nonRedundantFaces)
+    }
+    val sortedBlocks = nonRedundantBlocks.sortWith{ (left, right) =>
+      centroidCompare(left.centroid, right.centroid) }
+    val tolerance = 0.05
+
+    // If first block has satisfactory volume
+    if ((sortedBlocks.head.volume < (1.0 + tolerance)*desiredVolume) && 
+        (sortedBlocks.head.volume > (1.0 - tolerance)*desiredVolume) &&
+        (sortedBlocks.length == 2)) {
+      if (sortedBlocks.tail.head.volume <= (1.0 + tolerance)*desiredVolume) {
+        // If both blocks have satifactory volumes, prepend joint to joints
+        joint +: joints
+      } else {
+        // If last block isn't small enough
+        val remainingBlock = sortedBlocks.tail.head
```

Comment:
Figured it would fail more gracefully this way. It there anything bad about doing it this way?